src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs

```diff
@@ -87,6 +103,8 @@ protected override void OnEventCommand(EventCommandEventArgs command)
                 _ilBytesJittedCounter ??= new PollingCounter("il-bytes-jitted", this, () => System.Runtime.JitInfo.GetCompiledILBytes()) { DisplayName = "IL Bytes Jitted", DisplayUnits = "B" };
                 _methodsJittedCounter ??= new PollingCounter("methods-jitted-count", this, () => System.Runtime.JitInfo.GetCompiledMethodCount()) { DisplayName = "Number of Methods Jitted" };
                 _jitTimeCounter ??= new IncrementingPollingCounter("time-in-jit", this, () => System.Runtime.JitInfo.GetCompilationTime().TotalMilliseconds) { DisplayName = "Time spent in JIT", DisplayUnits = "ms", DisplayRateTimeScale = new TimeSpan(0, 0, 1) };
+
+                AppContext.LogSwitchValues(this);
```

Comment:
This is just logging values at a specific point in time (when the event source is enabled) - is that the desire?

I'm thinking of the possibility of scenarios like:
- Initial values do not have `SwitchX` set
- Event source is enabled, logging that `SwitchX` is not set
- App sets `SwitchX`
- Runtime (or app/library) reads `SwitchX` and is now configured a certain way

or

- Initial values do not have `SwitchX` set
- Runtime (or app/library) reads `SwitchX` and is now configured a certain way
- App sets `SwitchX`
- Event source is enabled, logging that `SwitchX` is set

Reply:
I think most of those are edge cases. The 90% usage is:
- Switch set in the entry point
- Switch set in runtime.config.json (which comes from the csproj)

Reply:
Definitely agree that those should not be the common case, but wanted to call it out.

> - Switch set in the entry point

I think part of my question is what does this end up looking like when tracing startup (where I'd expect the event source is enabled before the app's entry is is run)? Could it end up being the first scenario I had above?

Reply:
I'm not sure, @brianrob @josalem do event sources OnCommand run before Main executes when you attach the event source at startup (which not many do). 

@elinor-fung These are great questions BTW 👍🏾

Reply:
They definitely can, but not all will.  If the EventSource comes into existence before Main is called and it is told to turn on (e.g. EventPipe, ETW, or some EventListener tells it to), then it will trigger before Main.  By design, there isn't anything that requires that Main run before events can be emitted.  A good example here is FrameworkEventSource which often comes into existence before Main.

Reply:
To add to this, if you want to ensure that these can be emitted early, make sure that the EventSource that emits them comes into existence early.  That may already be the case, but if not, make sure it gets instantiated early, even if it doesn't log any events when instantiated.  Just the instantiation will allow it to be enabled immediately (at least by ETW, but shortly by EventPipe once it comes up).

Reply:
@elinor-fung, just read your first scenario - I think that the right way to handle a race between logging the current state and changes to switches is to dump the set of switches at the beginning of the tracing session as this code does.  Additionally, any time a switch is configured, it should be logged immediately.  This ensures that if switches change value or get set after the session is created, we see the change.  This, combined with instantiating the EventSource early will ensure that we don't miss any data.